pkgs/development/interpreters/perl/default.nix

```diff
@@ -1,20 +1,20 @@
 { config, lib, stdenv, fetchurl, pkgs, buildPackages, callPackage
-, enableThreading ? stdenv ? glibc, coreutils, makeWrapper
+, enableThreading ? stdenv.cc.isClang or (stdenv ? glibc), coreutils, makeWrapper
 }:
 
 with lib;
 
 # We can only compile perl with threading on platforms where we have a
-# real glibc in the stdenv.
+# real glibc in the stdenv or where clang is available.
 #
 # Instead of silently building an unthreaded perl if this is not the
-# case, we force callers to disableThreading explicitly, therefore
-# documenting the platforms where the perl is not threaded.
+# case, we force callers to explicitly set enableThreading to false,
+# therefore documenting the platforms where the perl is not threaded.
 #
 # In the case of stdenv linux boot stage1 it's not possible to use
 # threading because of the simpleness of the bootstrap glibc, so we
 # use enableThreading = false there.
```

Comment:
Makes sense to just override `enableThreading = false` in the bootstrap instead of implementing complicated logic in the package.

Reply:
Agree. One should assume perl can do multithreading on every platform, and then override where it's not possible. I don't know any platform where perl can't run multithreaded, given a sane libc.

Reply:
So this comment can be removed all together since it documents the logic that doesn't exist anymore.

We already have a comment in the linux stdenv bootstrap:
https://github.com/NixOS/nixpkgs/blob/349b471469fa9dd0293f26149a2860da3b6e3633/pkgs/stdenv/linux/default.nix#L209-L214

Reply:
The comment is correct. Bootstrap doesn't provide `glibc/libpthread_nonshared.a`, so the perl there has to be with threading disabled. The rest of the comment just states that in the final stdenv we'll get a different perl, this time with threading enabled. Actually, up to this PR, the comment was incorrect :)

Reply:
@knl By "this comment" I meant comment in the expression for `perl`. The comment in the implementation of `stdenv` is, obviously, is in the right place.

Reply:
@veprbl Ah, ok :) Makes more sense, removed the comment. Also, improved the commit message.